core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -497,11 +497,17 @@ private[spark] class BlockManagerInfo(
 
     updateLastSeenMs()
 
+    var blockExists = false
+    var originalMemSize: Long = 0
+    var originalDiskSize: Long = 0
+
     if (_blocks.containsKey(blockId)) {
       // The block exists on the slave already.
       val blockStatus: BlockStatus = _blocks.get(blockId)
       val originalLevel: StorageLevel = blockStatus.storageLevel
-      val originalMemSize: Long = blockStatus.memSize
+      originalMemSize = blockStatus.memSize
```

Comment:
BLocks of code even farther down can reuse these new values and need similar changes to the log comment